test_shell_app/test_shell_main.py

```diff
@@ -0,0 +1,55 @@
+import io
+import sys
+from unittest import TestCase
+from unittest.mock import patch, MagicMock
+
+from test_shell_app.shell_main import ShellMain
+
+INVALID_COMMAND = "NO_COMMAND"
+
+EXIT_COMMAND = "exit"
+
+
+class TestShellMain(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.shell_main = ShellMain()
+
+    def test_shell_main_operate_correctly(self):
+        output = io.StringIO()
+        backup = sys.stdout
+        sys.stdout = output
+        self.shell_main.show_init_message()
+        try:
+            self.assertEqual(self.shell_main.init_message, output.getvalue())
+        finally:
+            sys.stdout = backup
+
+    @patch.object(ShellMain, "get_user_input")
+    def test_shell_main_invalid_command(self, mock):
+        mock.side_effect = [INVALID_COMMAND, EXIT_COMMAND]
+        output = io.StringIO()
+        backup = sys.stdout
+        sys.stdout = output
+        self.shell_main.run()
+        try:
+            self.assertTrue(
+                self.shell_main.invalid_command_message in output.getvalue()
+            )
+        finally:
+            sys.stdout = backup
+
+    @patch.object(ShellMain, "get_user_input")
+    def test_shell_command_argument_check(self, mock):
+        test_command_map = {
+            "write": ("write 3 0xAAAABBBB", ("3", "0xAAAABBBB")),
+            "read": ("read 3", ("3",)),
+            "fullwrite": ("fullwrite 0xAAAABBBB", ("0xAAAABBBB",)),
+        }
+        for command, (full_command, expected_argument) in test_command_map.items():
+            with self.subTest(f"{command} test"):
+                self.shell_main.command_map[command] = MagicMock()
+                shell_command = self.shell_main.command_map[command]
+                mock.side_effect = [full_command, EXIT_COMMAND]
+                self.shell_main.run()
+                shell_command.assert_called_with(*expected_argument)
```

Comment:
Arrange, Act, Assert가 newline으로 구분되면 가독성이 좋아질 것 같습니다
```suggestion
                self.shell_main.command_map[command] = MagicMock()
                shell_command = self.shell_main.command_map[command]
                mock.side_effect = [full_command, EXIT_COMMAND]
                
                self.shell_main.run()
                
                shell_command.assert_called_with(*expected_argument)
```

Reply:
좋은 의견 같습니다! 추가하여 pr 올리고, 추후 개발에도 참고하겠습니다.